src/main/java/com/google/devtools/build/lib/analysis/constraints/RuleContextConstraintSemantics.java

```diff
@@ -830,6 +830,42 @@ private static void addSelectValuesToSet(BuildType.Selector<?> select, final Set
     }
   }
 
+  /**
+   * Provides information about a target's incompatibility.
+   *
+   * <p>After calling {@code checkForIncompatibility()}, the {@code isIncompatible} flag tells you
+   * whether the target is incompatible. If the target is incompatible, then {@code
+   * underlyingTarget} tells you which underlying target provided the incompatibility. For the vast
+   * majority of targets this is the same one passed to {@code checkForIncompatibility()}. In some
+   * instances like {@link OutputFileConfiguredTarget}, however, the {@code underlyingTarget} is the
+   * rule that generated the file.
+   */
+  public static final class IncompatibleCheckResult {
```

Comment:
Forwarding one more comment from @katre:

"Nit: This class will be difficult to debug because it has no toString method. Consider using AutoValue to create it and populate reasonable default toString/hashCode/equals methods."

There are other AutoValue classes in the code base that can show precedent.

Reply:
Ah yes, good idea. I'll get on this tonight.

Reply:
Done